src/cleanup_queue.h

```diff
@@ -0,0 +1,83 @@
+#ifndef SRC_CLEANUP_QUEUE_H_
+#define SRC_CLEANUP_QUEUE_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include <cstddef>
+#include <cstdint>
+#include <unordered_set>
+
+#include "memory_tracker.h"
+
+namespace node {
+
+class BaseObject;
+
+class CleanupQueue : public MemoryRetainer {
+ public:
+  typedef void (*Callback)(void*);
+
+  CleanupQueue() {}
+
+  // Not copyable.
+  CleanupQueue(const CleanupQueue&) = delete;
+
+  SET_MEMORY_INFO_NAME(CleanupQueue)
+  SET_NO_MEMORY_INFO()
```

Comment:
I think we can move the `env->ForEachBaseObject` loop in `Environment::BuildEmbedderGraph()` to the `MemoryInfo` method here and track the cleanup queue from the environment. That'll make the heapsnapshot closer to what the layout actually looks like.

Reply:
Sounds good to me. Updated.